containers/Dockerfile.buildenv

```diff
@@ -36,7 +36,8 @@ RUN apt-get update -y && apt-get install -y libc++-dev libc++1 libc++abi-dev \
 # Install ccache to cache compilations for reduced compile time, Doxygen
 # and coverxygen to check documentation coverage
 RUN apt-get install -y ccache doxygen python-pip && pip install coverxygen \
-    && pip install numpy && pip install scipy
+        && pip install numpy && pip install scipy \
+        && pip install --upgrade autopep8 flake8
```

Comment:
You can put all packages in one `pip install --upgrade` call, or is there a reason you don't want to do that?

Reply:
I prefer grouping related installs into a single command rather than having all installs in one command.

Reply:
For my own education: why are some `pip install` but others `pip install --upgrade` ?

Reply:
:man_shrugging: the internet told me to type those things so I did :P

Reply:
If the package is already installed, `--upgrade` will upgrade to the newest version. It's likely not needed since the docker container has basically no python packages installed anyway but I also don't think it's worth changing.